### PR TITLE
Viewer features and physics bugfix

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

This PR packages a few minor fixes for physics simulation and Viewer application features:
- Viewer:
  - `SPACE` to toggle physics simulation on/off
  - `'.'` to take a single simulation step when continuous simulation is toggled off.
  - Object adding returns allocated object IDs
  - Added objects are now set to agent orientation (aligned -Z forward local frames) instead of random orientation.
- [bugfix] Now correctly propagating configured collision margin to BulletRigidStage.

## How Has This Been Tested

Local tests.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/physics/bullet/BulletPhysicsManager.cpp [hunk 1]
```diff
@@ -161,6 +161,14 @@ void BulletPhysicsManager::stepPhysics(double dt) {
   int numSubStepsTaken =
       bWorld_->stepSimulation(dt, /*maxSubSteps*/ 10000, fixedTimeStep_);
   worldTime_ += numSubStepsTaken * fixedTimeStep_;
+
+  // Manually sync the motionstates for all DYNAMIC objects. KINEMATIC and
+  // STATIC sync should always be one-way.
+  for (auto& objectItr : existingObjects_) {
```
Comment (eundersander) [inline src/esp/physics/bullet/BulletPhysicsManager.cpp:167]: Isn't Bullet supposed to do this automatically through the motion state interface we give it? Have you investigated why this isn't happening correctly in Bullet? This feels like a band-aid.

Comment (aclegg3) [reply src/esp/physics/bullet/BulletPhysicsManager.cpp]: Fair point. Digging deeper I did find another option. I'll open another PR to iterate on this specific issue.

File: src/utils/viewer/viewer.cpp [hunk 2]
```diff
@@ -88,14 +88,14 @@ class Viewer : public Mn::Platform::Application {
   void keyPressEvent(KeyEvent& event) override;
 
   // Interactive functions
-  void addObject(const std::string& configHandle);
-  void addObject(int objID);
+  int addObject(const std::string& configHandle);
+  int addObject(int objID);
 
   // add template-derived object
```
Comment (bigbike) [inline src/utils/viewer/viewer.cpp:94]: consider specifying the return value in the comments.

File: src/utils/viewer/viewer.cpp [hunk 3]
```diff
@@ -382,42 +391,46 @@ Viewer::Viewer(const Arguments& arguments)
...
+int Viewer::addObject(const std::string& objectAttrHandle) {
   // Relative to agent bodynode
   Mn::Matrix4 T = agentBodyNode_->MagnumObject::transformationMatrix();
   Mn::Vector3 new_pos = T.transformPoint({0.1f, 1.5f, -2.0f});
 
-  int physObjectID = simulator_->addObjectByHandle(configFile);
+  int physObjectID = simulator_->addObjectByHandle(objectAttrHandle);
   simulator_->setTranslation(new_pos, physObjectID);
-  simulator_->setRotation(esp::core::randomRotation(), physObjectID);
+  simulator_->setRotation(Mn::Quaternion::fromMatrix(T.rotationNormalized()),
```
Comment (bigbike) [inline src/utils/viewer/viewer.cpp:407]: > set to agent orientation

I would like to learn a bit more about this design choice. Why not set it to align the stage orientation?

Comment (aclegg3) [reply src/utils/viewer/viewer.cpp:407]: We could do that also. Just aiming for something consistent here. 

My reasoning: We are setting position from agent transform, so do the same for orientation. This way you always know that the object's local forward is aligned with the camera forward when it is added. 

Another choice: we could apply no transformation to added objects by default and provide another function to set an object transform (with orientation optional) relative to the agent as a follow-up operation. Do you have a preference?

**Review by jturner65 — COMMENTED** [review]
LGTM - if it passes is ships! :)

**Review by jturner65 — APPROVED** [review]

**Pull request merged**